src/widgets/TooltipWidget.cpp

```diff
@@ -25,44 +19,52 @@ TooltipWidget::TooltipWidget(BaseWidget *parent)
     : BaseWindow({BaseWindow::TopMost, BaseWindow::DontFocus,
                   BaseWindow::DisableLayoutSave},
                  parent)
-    , displayImage_(new QLabel())
-    , displayText_(new QLabel())
+    , displayImage_(new QLabel(this))
+    , displayText_(new QLabel(this))
 {
     this->setStyleSheet("color: #fff; background: rgba(11, 11, 11, 0.8)");
     this->setAttribute(Qt::WA_TranslucentBackground);
-    //this->setWindowOpacity(0.8);
-    this->updateFont();
+    this->setWindowFlag(Qt::WindowStaysOnTopHint, true);
+
     this->setStayInScreenRect(true);
 
-    displayImage_->hide();
     displayImage_->setAlignment(Qt::AlignHCenter);
     displayImage_->setStyleSheet("background: transparent");
+
     displayText_->setAlignment(Qt::AlignHCenter);
-    displayText_->setText("tooltip text");
     displayText_->setStyleSheet("background: transparent");
-    auto layout = new QVBoxLayout();
+
+    auto *layout = new QVBoxLayout(this);
+    layout->setSizeConstraint(QLayout::SetFixedSize);
     layout->setContentsMargins(10, 5, 10, 5);
     layout->addWidget(displayImage_);
     layout->addWidget(displayText_);
     this->setLayout(layout);
 
-    this->fontChangedConnection_ = getFonts()->fontChanged.connect([this] {
+    this->connections_.managedConnect(getFonts()->fontChanged, [this] {
         this->updateFont();
     });
-}
+    this->updateFont();
```

Comment:
Can you verify that `updateFont` isn't called twice here? `connect` functions to signals and settings should, by default, call the callback once unless another parameter is set to `false` (can't remember its name atm)

EDIT: I realize `updateFont` was already called like this here, you've just moved the call down so feel free to just add a comment here if you don't want to handle this in this PR

Reply:
confirmed its not called twice, only settings signals have autoInvoke from what i see, so will leave it in current form in this PR